tests/integration/base.py

```diff
@@ -0,0 +1,299 @@
+            )
+            self.assertEqual(
+                self.invoke_output["resources"], ["arn:aws:ec2:us-east-1:123456789012:instance/i-abcd1111"],
+            )
+            self.assertEqual(self.invoke_output["source"], "aws.ec2")
+            self.assertEqual(self.invoke_output["account"], "123456789012")
+            self.assertEqual(self.invoke_output["region"], "us-east-1")
+
+    class HelloWorldExclamationBuildInvokeBase(BuildInvokeBase):
+        def _test_local_invoke(self):
+            super()._test_local_invoke()
+            self.assertEqual(self.invoke_output["statusCode"], 200)
+            self.assertEqual(json.loads(self.invoke_output["body"]), {"message": "Hello World!"})
+
+    class SimpleHelloWorldBuildInvokeBase(BuildInvokeBase):
+        def _test_local_invoke(self):
+            super()._test_local_invoke()
+            self.assertEqual(self.invoke_output["statusCode"], 200)
+            self.assertEqual(json.loads(self.invoke_output["body"]), {"message": "hello world"})
+
+    class QuickStartWebBuildInvokeBase(BuildInvokeBase):
+        function_id_by_event = {
+            "event-get-all-items.json": "getAllItemsFunction",
+            "event-get-by-id.json": "getByIdFunction",
+            "event-post-item.json": "putItemFunction",
+        }
+
+    # Unit Tests
+    class UnitTestBase(IntegBase):
+        # by default, the code is not in a nested directory
+        code_directories: List[str] = ["."]
+
+        @abstractmethod
+        def _test_install(self, code_directory: str):
+            pass
+
+        @abstractmethod
+        def _test_unit_tests(self, code_directory: str):
+            pass
+
+        def test_unit_tests(self):
+            self._test_init_template()
+            for code_directory in self.code_directories:
+                self._test_install(code_directory)
+                self._test_unit_tests(code_directory)
+
+    class NodejsUnitTestBase(UnitTestBase):
+        def _test_install(self, code_directory: str):
+            cmdlist = [
+                "npm",
+                "i",
+            ]
+            LOG.info(cmdlist)
+            result = run_command(cmdlist, Path(self.cwd, code_directory))
+            self.assertRegex(
+                result.stdout.decode(), r"added \d+ packages from \d+ contributors and audited \d+ packages",
+            )
+            self.assertIn(
+                "found 0 vulnerabilities", result.stdout.decode(),
+            )
+
+        def _test_unit_tests(self, code_directory: str):
+            cmdlist = [
+                "npm",
+                "test",
+            ]
+            LOG.info(cmdlist)
+            result = run_command(cmdlist, Path(self.cwd, code_directory))
+            self.assertIn("pass", result.stdout.decode() + result.stderr.decode())
+            self.assertNotIn("fail", result.stdout.decode() + result.stderr.decode())
+
+    class PythonUnitTestBase(UnitTestBase):
+        python_executable = "python"
```

Comment:
should we be more cognizant of the version?

say python2.7, python3.6, python3.7 and python3.8

Reply:
good idea, done